core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushMergeWriterUpdateIntoConnector.java

```diff
@@ -0,0 +1,171 @@
+                            new TrinoException(GENERIC_INTERNAL_ERROR, "Catalog not set")),
+                            schemaTableName.getSchemaName(),
+                            schemaTableName.getTableName()))
+                    .orElseThrow(() -> new TrinoException(GENERIC_INTERNAL_ERROR, "Cannot get table handle"));
+            Map<String, ColumnHandle> columnHandles = plannerContext.getMetadata()
+                    .getColumnHandles(session, tableHandle);
+
+            List<String> orderedColumnNames = mergeWriter.getTarget().getMergeParadigmAndTypes().getColumnNames();
+            List<? extends Node> mergeAssignments = project.getAssignments().get(mergeProcessor.getMergeRowSymbol()).getChildren();
+            ImmutableMap.Builder<ColumnHandle, Constant> assignmentsBuilder = ImmutableMap.builder();
+            for (int i = 0; i < orderedColumnNames.size(); i++) {
+                String columnName = orderedColumnNames.get(i);
+                Node assigmentNode = mergeAssignments.get(i);
+                if (isSymbol(assigmentNode)) {
+                    // the column is not updated
+                    continue;
+                }
+                if (!isConstant(assigmentNode)) {
+                    return Result.empty();
+                }
+
+                ConnectorExpressionTranslator.ConnectorExpressionTranslation expressionTranslation = ConnectorExpressionTranslator.translateConjuncts(
+                        context.getSession(),
+                        ((Expression) assigmentNode),
+                        context.getSymbolAllocator().getTypes(),
+                        plannerContext,
+                        typeAnalyzer);
+
+                ConnectorExpression connectorExpression = expressionTranslation.connectorExpression();
+                // we don't support any expressions in update statements yet, only constants
+                // also in case of UPDATE ALL statement - row represented as base64 call
+                if (!(connectorExpression instanceof Constant)) {
+                    return Result.empty();
+                }
+                assignmentsBuilder.put(columnHandles.get(columnName), (Constant) connectorExpression);
+            }
+            Map<ColumnHandle, Constant> assignments = assignmentsBuilder.buildOrThrow();
+            if (assignments.isEmpty()) {
+                return Result.empty();
+            }
+
+            return metadata.applyUpdate(context.getSession(), tableScan.getTable(), assignments)
+                    .map(newHandle -> new TableUpdateNode(
+                            context.getIdAllocator().getNextId(),
+                            newHandle,
+                            getOnlyElement(node.getOutputSymbols())))
+                    .map(Result::ofPlanNode)
+                    .orElseGet(Result::empty);
+        }
+        catch (RuntimeException e) {
```

Comment:
What is that? I don't think we should swallow any generic exception like NPE here or TrinoException that should be propagated to the user.